src/main/java/uk/gov/companieshouse/acsp/controllers/AcspController.java

```diff
@@ -48,4 +47,16 @@ public ResponseEntity<Object> getAcspData(@PathVariable(TRANSACTION_ID_KEY) Stri
         }
     }
 
+    @GetMapping("/acsp-api/user/{id}/application")
+    public ResponseEntity<Object> checkHasApplication(@PathVariable("id") String id,
+                                                @RequestHeader(value = ERIC_ACCESS_TOKEN) String requestId){
+        LOGGER.info("received request to check for user applications");
+        int acspCount = acspService.getAcspApplicationCount(id);
```

Comment:
You can do this logic in the service layer only Controller can return the response as it received from service